lib/src/client.dart

```diff
@@ -2058,14 +2058,22 @@ class Client extends MatrixApi {
   /// Resets all settings and stops the synchronisation.
```

Comment:
Hm, I don't see _migrateFromLegacyDatabase calling .clear(), when is clear expected to be called?

Reply:
It depends on the implementation but calling `delete` should have been enough. We're calling `clear` just to be sure.

In the case of hive db, we should always call `clear` even if we call delete because it doesn't really delete all the boxes. We could also just fix the `delete` for hive db implementation but that'd mean if someone uses a custom db and does this exact mistake, it'd be hard to track.

I chose to go this way because we already do the same for `database` in logout.

Reply:
Yes but _migrateFromLegacyDatabase only calls delete, so when does the clear get called? 🤔

Reply:
It doesn't get called and that is the issue. So, I called it in there as well.

Reply:
found it